redfish-core/lib/license_service.hpp

```diff
@@ -0,0 +1,693 @@
+        asyncResp->res.jsonValue["@odata.type"] =
+            "#LicenseCollection.LicenseCollection";
+        asyncResp->res.jsonValue["@odata.id"] =
+            "/redfish/v1/LicenseService/Licenses";
+        asyncResp->res.jsonValue["Name"] = "License Collection";
+
+        getLicenseEntryCollection(asyncResp);
+    });
+
+    BMCWEB_ROUTE(app, "/redfish/v1/LicenseService/Licenses")
+        .privileges({{"ConfigureManager"}})
+        .methods(boost::beast::http::verb::post)(
+            [](const crow::Request& req,
+               const std::shared_ptr<bmcweb::AsyncResp>& asyncResp) {
+        std::string licenseString;
+        if (!json_util::readJsonAction(req, asyncResp->res, "LicenseString",
+                                       licenseString))
+        {
+            return;
+        }
+
+        if (licenseString.empty())
+        {
+            messages::propertyMissing(asyncResp->res, "LicenseString");
+        }
+
+        // Only allow one License install at a time
+        if (getLicenseActivationStatusMatch() != nullptr)
+        {
+            messages::resourceInUse(asyncResp->res);
+            return;
+        }
+
+        std::shared_ptr<boost::asio::steady_timer> timeout =
+            std::make_shared<boost::asio::steady_timer>(
+                crow::connections::systemBus->get_io_context());
+        timeout->expires_after(std::chrono::seconds(20));
+        crow::connections::systemBus->async_method_call(
+            [timeout, asyncResp,
+             licenseString](const boost::system::error_code ec) {
+            if (ec)
+            {
+                BMCWEB_LOG_ERROR("LicenseString resp_handler got error {}", ec);
+                messages::internalError(asyncResp->res);
+                return;
+            }
+
+            auto timeoutHandler =
+                [asyncResp, timeout](const boost::system::error_code errCode) {
+                resetLicenseString(asyncResp);
+                if (errCode)
+                {
+                    if (errCode != boost::asio::error::operation_aborted)
+                    {
+                        BMCWEB_LOG_ERROR("Async_wait failed {}", errCode);
+                        messages::internalError(asyncResp->res);
+                        return;
+                    }
+                }
+                else
+                {
+                    messages::serviceTemporarilyUnavailable(asyncResp->res,
+                                                            "60");
+                    BMCWEB_LOG_ERROR("Timed out waiting for HostInterface to "
+                                     "serve license upload request");
+                    return;
+                }
+            };
+
+            timeout->async_wait(timeoutHandler);
+
+            auto callback = [asyncResp, timeout,
+                             licenseString](sdbusplus::message::message& m) {
+                BMCWEB_LOG_DEBUG("Response Matched "); //, m.get());
```

Comment:
Why the comment on m.get()? Can you just remove ?

Reply:
done